src/charm.py

```diff
@@ -86,6 +202,13 @@ def _on_leader_elected(self, _) -> None:
         data.setdefault("postgres-password", self._new_password())
         data.setdefault("replication-password", self._new_password())
 
+    def _get_ips_to_remove(self) -> List[str]:
+        """List the IPs that were part of the cluster but departed."""
+        old = self.members_ips
+        current = self._units_ips
+        endpoints_to_remove = list(set(old) - set(current))
```

Comment:
I'm wondering why don't treat all of this structures as sets. So you avoid this conversion/reconversion

Reply:
Good suggestion. Changed in https://github.com/canonical/postgresql-operator/pull/10/commits/bf34427e76fade29e4297b0dc5ae018257b0b69c.